analytics-integrations/kahuna/src/main/java/com/segment/analytics/internal/integrations/KahunaIntegration.java

```diff
@@ -60,25 +89,119 @@
   @Override public void identify(IdentifyPayload identify) {
     super.identify(identify);
 
-    String username = identify.traits().username();
-    KahunaAnalytics.setUsernameAndEmail(isNullOrEmpty(username) ? identify.userId() : username,
-        identify.traits().email());
-
-    KahunaAnalytics.setUserCredential(USERNAME_KEY, identify.traits().username());
-    KahunaAnalytics.setUserCredential(EMAIL_KEY, identify.traits().email());
-    KahunaAnalytics.setUserCredential(FACEBOOK_KEY, identify.traits().getString("facebook"));
-    KahunaAnalytics.setUserCredential(TWITTER_KEY, identify.traits().getString("twitter"));
-    KahunaAnalytics.setUserCredential(LINKEDIN_KEY, identify.traits().getString("linkedin"));
-
-    KahunaAnalytics.setUserAttributes(identify.traits().toStringMap());
+    Traits traits = identify.traits();
```

Comment:
It looks like we are missing a call here, we should be tracking the userID also with the following:

KahunaAnalytics.setUserCredential(USER_ID_KEY, identify.userId());

Kahuna tracks UserId as "user_id" and Segment tracks as "userId" so here we track it explicitly outside of the traits for loop.